### PR TITLE
[styles] Add Tehran Metro icons and bounding box

### PR DESCRIPTION
**Pull request by claudiush**
Adding the Tehran metro icon files medium and small as well as the bounding box information for the metro network.

This Patch does not yet include:

- night style icon files: Can you share instructions on best practices for creating these? I see most of them are just dimmed darker, while e.g. London Tube looks identical to the day style
- Update to the file data/mapcss-mapping.csv to add Tehran there as I'm not sure this file gets manually updated or automatically generated so that the unique IDs are preserved.

### CODE REVIEW

File: generator/osm2type.cpp [hunk 1]
```diff
@@ -325,7 +325,7 @@ void LeaveLongestTypes(vector<generator::TypeStrings> & matchedTypes)
 
 void MatchTypes(OsmElement * p, FeatureBuilderParams & params, function<bool(uint32_t)> const & filterType)
 {
-  auto static const rules = generator::ParseMapCSS(GetPlatform().GetReader(MAPCSS_MAPPING_FILE));
+  auto static const rules = generator::ParseMapCSS(GetPlatform().GetReader("mapcss-mapping.csv"));
```
Comment (vng) [inline generator/osm2type.cpp:328]: What went wrong here?

File: generator/osm2type.cpp [hunk 2]
```diff
@@ -423,7 +423,7 @@ string MatchCity(OsmElement const * p)
       {"montreal", {-73.995802, 45.398482, -73.474295, 45.70479}},
       {"moscow", {36.9964599609, 55.3962717136, 38.1884765625, 56.1118730004}},
       {"mumbai", {72.7514648437, 18.8803004445, 72.9862976074, 19.2878132403}},
-      {"munchen", {11.3433837891, 47.9981928195, 11.7965698242, 48.2730267576}},
+      {"munchen", {11.3433837891, 47.9981928195, 11.7965698242, 48.2530267576}},
```
Comment (vng) [inline generator/osm2type.cpp:426]: Why?

File: generator/osm2type.cpp [hunk 3]
```diff
@@ -461,6 +461,7 @@ string MatchCity(OsmElement const * p)
       {"taoyuan", {110.8471,28.4085,111.6109,29.4019}}, 
       {"tashkent", {69.12171, 41.163421, 69.476967, 41.398638}},
       {"tbilisi", {44.596922, 41.619315, 45.019694, 41.843421}},
+      {"tehran", {50.6575}},
```
Comment (vng) [inline generator/osm2type.cpp:464]: Expected 4 floats, like Rect

**Review by vng — COMMENTED** [review]
We can take your icons and make a good PR for that.
Just let us know the bounding box for Tehran.

File: generator/osm2type.cpp (hunk 1 above)
Comment (claudiush) [reply generator/osm2type.cpp]: Seems I based this off an older base version and not the head of the file which also explains the comment change below, too

File: generator/osm2type.cpp (hunk 2 above)
Comment (claudiush) [reply generator/osm2type.cpp]: See earlier comment. I made my changes not from latest head file. Re-submitted

File: generator/osm2type.cpp (hunk 3 above)
Comment (claudiush) [reply generator/osm2type.cpp]: Fixed. Should be 50.6575,35.353216,52.007904,35.974672

**Review by claudiush — COMMENTED** [review]
I've re-submitted corrected osm2type.cpp through this github.com interface but not sure where it ended up :)

**Comment by vng** [comment]
Thanks!
Complete PR: https://github.com/organicmaps/organicmaps/pull/5611

**Pull request closed without merging**